pycontrails/models/aircraft_performance.py

```diff
@@ -4,34 +4,405 @@
+        air_temperature: npt.NDArray[np.float_],
+        time: npt.NDArray[np.datetime64] | None,
+        true_airspeed: npt.NDArray[np.float_] | float | None,
+        aircraft_mass: npt.NDArray[np.float_] | float,
+        engine_efficiency: npt.NDArray[np.float_] | float | None,
+        fuel_flow: npt.NDArray[np.float_] | float | None,
+        thrust: npt.NDArray[np.float_] | float | None,
+        q_fuel: float,
+        **kwargs: Any,
+    ) -> AircraftPerformanceData:
+        r"""
+        Calculate aircraft performance along a trajectory.
+
+        When ``time`` is not None, this method should be used for a single flight
+        trajectory. Waypoints are coupled via the ``time`` parameter.
+
+        This method computes the rate of climb and descent (ROCD) to determine
+        flight phases: "cruise", "climb", and "descent". Performance metrics
+        depend on this phase.
+
+        When ``time`` is None, this method can be used to simulate flight performance
+        over an arbitrary sequence of flight waypoints by assuming nominal flight
+        characteristics. In this case, each point is treated independently and
+        all points are assumed to be in a "cruise" phase of the flight.
+
+        Parameters
+        ----------
+        aircraft_type : str
+            Used to query the underlying model database for aircraft engine parameters.
+        altitude_ft : npt.NDArray[np.float_]
+            Altitude at each waypoint, [:math:`ft`]
+        air_temperature : npt.NDArray[np.float_]
+            Ambient temperature for each waypoint, [:math:`K`]
+        time: npt.NDArray[np.datetime64] | None
+            Waypoint time in ``np.datetime64`` format. If None, only drag force
+            will is used in thrust calculations (ie, no vertical change and constant
+            horizontal change). In addition, aircraft is assumed to be in cruise.
+        true_airspeed : npt.NDArray[np.float_] | float | None
+            True airspeed for each waypoint, [:math:`m s^{-1}`].
+            If None, a nominal value is used.
+        aircraft_mass : npt.NDArray[np.float_] | float
+            Aircraft mass for each waypoint, [:math:`kg`].
+        engine_efficiency : npt.NDArray[np.float_] | float | None
+            Override the engine efficiency at each waypoint.
+        fuel_flow : npt.NDArray[np.float_] | float | None
+            Override the fuel flow at each waypoint, [:math:`kg s^{-1}`].
+        thrust : npt.NDArray[np.float_] | float | None
+            Override the thrust setting at each waypoint, [:math: `N`].
+        q_fuel : float
+            Lower calorific value (LCV) of fuel, [:math:`J \ kg_{fuel}^{-1}`].
+        **kwargs : Any
+            Additional keyword arguments to pass to the model.
+
+        Returns
+        -------
+        AircraftPerformanceData
+            Derived performance metrics at each waypoint.
+        """
+
+    def ensure_tas_on_source(self) -> None:
```

Comment:
I find the `tas` label is not universal, so I'd probably make this way verbose as `ensure_true_airspeed` or `ensure_true_airspeed_on_source`

Reply:
2c70513